### PR TITLE
Fix(tsql): include catalog and db in create if not exists

### PR DESCRIPTION
**Pull request by treysp**

### CODE REVIEW

File: sqlglot/dialects/tsql.py [hunk 1]
```diff
@@ -667,7 +667,19 @@ def create_sql(self, expression: exp.Create) -> str:
                 if kind == "SCHEMA":
                     sql = f"""IF NOT EXISTS (SELECT * FROM information_schema.schemata WHERE schema_name = {identifier}) EXEC('{sql}')"""
                 elif kind == "TABLE":
-                    sql = f"""IF NOT EXISTS (SELECT * FROM information_schema.tables WHERE table_name = {identifier}) EXEC('{sql}')"""
+                    assert table
```
Comment (GeorgeSittas) [inline sqlglot/dialects/tsql.py:670]: Shouldn't we also apply this fix to the other cases for `kind`?

File: sqlglot/dialects/tsql.py [hunk 2]
```diff
@@ -667,7 +667,19 @@ def create_sql(self, expression: exp.Create) -> str:
...
+                    where = f"table_name = {self.sql(exp.Literal.string(table.name))}"
```
Comment (GeorgeSittas) [inline sqlglot/dialects/tsql.py:671]: I don't think the `exp.Literal.string` adds anything here, you can simply do `f"table_name = {table.name}"`.

File: sqlglot/dialects/tsql.py (hunk 1 above)
Comment (treysp) [reply sqlglot/dialects/tsql.py:670]: it's selecting from `sys.indexes` using an object id, and that table doesn't include schema/catalog names

Comment (treysp) [reply sqlglot/dialects/tsql.py:670]: sorry, that was for index -- i'll add it for schema

Comment (treysp) [reply sqlglot/dialects/tsql.py:670]: and schema can only come in unqualified, so identifier will always work

Comment (GeorgeSittas) [reply sqlglot/dialects/tsql.py:670]: i see, makes sense

**Review by GeorgeSittas — APPROVED** [review]

File: sqlglot/dialects/tsql.py (hunk 2 above)
Comment (tobymao) [inline sqlglot/dialects/tsql.py:671]: where = exp.column("table_name").eq(table.name)

File: sqlglot/dialects/tsql.py [hunk 3]
```diff
@@ -667,7 +667,19 @@ def create_sql(self, expression: exp.Create) -> str:
...
+                    where = (
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:672]: where = table_name.and_(...)

**Pull request merged**